### ISSUE TITLE
failed to create bridge "cni0": could not add "cni0": operation not supported

### ISSUE THREAD

**Issue by ChenQi1989** [containernetworking/plugins#370]
I'm trying to setup k8s using flannel. But I'm blocked by this error. Hope someone could give me some suggestion.

I used the following commands:
1. swapoff -a
2. kubeadm init --pod-network-cidr=10.244.0.0/16
3. sysctl net.bridge.bridge-nf-call-iptables=1
4. mkdir -p $HOME/.kube
    cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
    chown $(id -u):$(id -g) $HOME/.kube/config
5. kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml

I also added some output in kubelet and link_linux.go to help debugging. But I'm now blocked here. Below are the related logs. Any idea?

Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: I0813 03:30:39.018857    9968 raw_exec.go:40] chenqi: raw_exec.ExecPlugin &{/opt/cni/bin/flannel [/opt/cni/bin/flannel] [PATH=/usr/local/sbin:/usr/local/bin:/usr
/sbin:/usr/bin:/sbin:/bin INVOCATION_ID=ed7608a7f620418695c029825e3d374d CNI_ARGS=IgnoreUnknown=1;K8S_POD_NAMESPACE=kube-system;K8S_POD_NAME=coredns-5f7fc64c95-x4l8t;K8S_POD_INFRA_CONTAINER_ID=ce1302be7aedaf89
32763b461a287fa5437a3889580fa615c7422f43f65d4c54 KUBELET_KUBEADM_ARGS=--cgroup-driver=systemd --network-plugin=cni --pod-infra-container-image=k8s.gcr.io/pause:3.1 --resolv-conf=/run/systemd/resolve/resolv.con
f CNI_IFNAME=eth0 LANG=C JOURNAL_STREAM=8:3119020 KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf CNI_CONTAINERID=ce1302be7aedaf8
932763b461a287fa5437a3889580fa615c7422f43f65d4c54 CNI_NETNS=/proc/12016/ns/net KUBELET_CONFIG_ARGS=--config=/var/lib/kubelet/config.yaml KUBELET_EXTRA_ARGS=-v 4 CNI_COMMAND=ADD CNI_PATH=/opt/cni/bin]  {"cniVer
sion":"","delegate":{"hairpinMode":true,"isDefaultGateway":true},"name":"cbr0","type":"flannel"}  %!s(*os.File=&{0xc00009e120}) [] %!s(*syscall.SysProcAttr=<nil>) %!s(*os.Process=<nil>) <nil> %!s(*context.empt
yCtx=0xc000052038) <nil> %!s(bool=false) [] [] [] [] %!s(chan error=<nil>) %!s(chan struct {}=<nil>)}
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: 2019/08/13 03:30:39 chenqi: link_linux.go: link type is *netlink.Bridge
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: 2019/08/13 03:30:39 chenqi: link_linux.go: after req.AddData
Aug 13 03:30:39 yocto-k8s-master systemd[1]: Created slice libcontainer_12120_systemd_test_default.slice.
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: I0813 03:30:39.030635    9968 factory.go:177] Factory "docker" was unable to handle container "/libcontainer_12120_systemd_test_default.slice"
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: I0813 03:30:39.030661    9968 factory.go:166] Error trying to work out if we can handle /libcontainer_12120_systemd_test_default.slice: /libcontainer_12120_syste
md_test_default.slice not handled by systemd handler
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: I0813 03:30:39.030668    9968 factory.go:177] Factory "systemd" was unable to handle container "/libcontainer_12120_systemd_test_default.slice"
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: I0813 03:30:39.030678    9968 factory.go:170] Factory "raw" can handle container "/libcontainer_12120_systemd_test_default.slice", but ignoring.
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: I0813 03:30:39.030689    9968 manager.go:908] ignoring container "/libcontainer_12120_systemd_test_default.slice"
Aug 13 03:30:39 yocto-k8s-master systemd[1]: Removed slice libcontainer_12120_systemd_test_default.slice.
Aug 13 03:30:39 yocto-k8s-master systemd-udevd[12091]: link_config: autonegotiation is unset or enabled, the speed and duplex are not writable.
Aug 13 03:30:39 yocto-k8s-master systemd-udevd[12091]: link_config: could not get ethtool features for cni0
Aug 13 03:30:39 yocto-k8s-master systemd-udevd[12091]: Could not set offload features of cni0: No such device
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: 2019/08/13 03:30:39 chenqi: link_linux.go: req.Execute(unix.NETLINK_ROUTE, 0) failed
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: I0813 03:30:39.041036    9968 raw_exec.go:51] chenqi: in pluginErr: error: exit status 1; output = {
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]:     "code": 100,
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]:     "msg": "failed to create bridge \"cni0\": could not add \"cni0\": operation not supported"
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: }
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: I0813 03:30:39.041077    9968 raw_exec.go:59] chenqi: pluginErr returning error message: failed to create bridge "cni0": could not add "cni0": operation not supported
Aug 13 03:30:39 yocto-k8s-master kubelet[9968]: E0813 03:30:39.041085    9968 cni.go:338] Error adding kube-system_coredns-5f7fc64c95-x4l8t/ce1302be7aedaf8932763b461a287fa5437a3889580fa615c7422f43f65d4c54 to network flannel/cbr0: failed to create bridge "cni0": could not add "cni0": operation not supported

**Comment by ChenQi1989** [comment]
This seems to be a regression. As if I use v0.7.5, there's no such problem.

commit a62711a5da7a2dc2eb93eac47e103738ad923fd6 (tag: v0.7.5, origin/v0.7)

**Comment by ChenQi1989** [comment]
I've tracked down to the following commit. I think it caused such regression.

commit dc536993e2b390eea0dbcafead5550495a810e36
Author: Sebastian Sch <sebassch@gmail.com>
Date:   Sun Nov 18 15:43:35 2018 +0200

    vendor folder bump.

**Comment by squeed** [comment]
Interesting
The latest release. v0.7.6, should fix this.

**Comment by ChenQi1989** [comment]
@squeed v0.7.5 works for me. Checking the v0.7 branch, I think v0.7.6 should also work.
The problem I encountered is about origin/master branch.

**Comment by dcbw** [comment]
@ChenQi1989 git master has the same commit as v0.7.6 though...

**Comment by ChenQi1989** [comment]
@dcbw I mean things are working well on v0.7 branch, including v0.7.5 and v0.7.6. But they are working well on master branch. I tracked down to the following commit.
"""
commit dc536993e2b390eea0dbcafead5550495a810e36
Author: Sebastian Sch <sebassch@gmail.com>
Date:   Sun Nov 18 15:43:35 2018 +0200

    vendor folder bump.
"""

But I don't know the root cause.
I tried k8s 1.15.1 + cni plugin master + flannel setup, the problem could be reproduced.

**Comment by ChenQi1989** [comment]
@dcbw Sorry for typo above. I mean the master branch is NOTworking well.

**Comment by Promaethius** [comment]
Having the same issue on master and v0.8.2

**Comment by squeed** [comment]
I suspect I know the issue.

Can you try building the binaries manually with this line commented out? https://github.com/containernetworking/plugins/blob/master/plugins/main/bridge/bridge.go#L224

**Comment by carlosedp** [comment]
I'm seeing this behavior on a Raspberry Pi with plugins v0.8.2.

When I reverted to v0.7.5 it worked.

```
pi@raspberrypi:~/faasd $ uname -a
Linux raspberrypi 4.19.75-v7+ #1270 SMP Tue Sep 24 18:45:11 BST 2019 armv7l GNU/Linux
pi@raspberrypi:~/faasd $ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

cc. @alexellis

**Comment by carlosedp** [comment]
Confirming that commenting the line as pointed by @squeed fixes the problem and the bridge interface is created.

**Comment by alexellis** [comment]
Does `git blame` show why it was added and how it was tested in the commit message? I'd be curious to know if this is an issue that affects x86_64, or just RPi/armhf.

**Comment by alexellis** [comment]
@bboreham cc

**Comment by carlosedp** [comment]
Maybe @SchSeba could also help out figuring if it's something missing on Kernel side.

**Comment by l1key0u** [comment]
It is work on k8s master, after use this command.
kubectl taint nodes --all node-role.kubernetes.io/master-

**Comment by l1key0u** [comment]
I try to reinstall k8s, it is work!
Maybe lost somethings.

**Comment by l1key0u** [comment]
> I try to reinstall k8s, it is work!
> Maybe lost somethings.

If the node is not running coredns, cni0 will not be created.

**Comment by bboreham** [comment]
Closed by #434 

> If the node is not running coredns, cni0 will not be created.

@l1key0u sounds unlikely.  If you're still having problems with the Flannel CNI plugin please open a new issue and supply full details.
